aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -138,6 +148,109 @@ module aptos_framework::coin {
     /// Capability required to burn coins.
     struct BurnCapability<phantom CoinType> has copy, store {}
 
+    /// The mapping between coin and fungible asset.
+    struct CoinConversionMap has key {
+        coin_to_fungible_asset_map: Table<TypeInfo, address>,
```

Comment:
can we add a reverse map `fungible_asset_to_coin_map` and a view function so that devs can also access TypeInfo through metadata?
```
#[view]
public fun paired_coin(metadata: Object<Metadata>): TypeInfo { ... }
```
I believe this is useful